src/compute-client/src/plan/transform/api.rs

```diff
@@ -9,17 +9,19 @@
 
 //! Utilities for transformation of [crate::plan::Plan] structures.
 
+use std::collections::BTreeSet;
+
 use mz_ore::stack::RecursionLimitError;
 
 use crate::plan::interpret::{BoundedLattice, FoldMut, Interpreter};
 use crate::plan::Plan;
 
 /// The type of configuration options passed to all [Transform::transform] calls
 /// as an immutable reference.
-pub type TransformConfig = ();
+pub type TransformConfig<C> = BTreeSet<C>;
```

Comment:
nit: I'm not sure what benefit we are getting from making `TransformConfig` generic in `C`. I think that for every version of the code all `Transform` instances would need to to get a `TransformConfig` instance with the same `C` value because we would want to pass a shared config (similar to `TransformArgs` in the `transform` crate).

Reply:
Ah, I see... I interpreted this as the configuration that you'd like to pass to a specific `Transform` and not a shared configuration for all `Transform`s. Right now, I only need to represent the former. What do you think would be a good representation in this case?

Reply:
> Right now, I only need to represent the former.

If by "former" you mean

> the configuration that you'd like to pass to a specific `Transform`

The recommended way is to do as suggested above and:

1. Re-define `TransformConfig` as a general purpose struct.
2. Project fields from that struct when you construct the interpreter.

Alternatively we can remove this part of the API for now and use some explicit assembly, but I feel that we will need it as soon as we want to run more than one transform.

Reply:
Cool, I revisited that now and created `TransformConfig` as a struct like you suggested.